src/pages/ItemList/Components/Filter.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import { useState } from 'react';
+
+const Filter = () => {
+  const [isSelectedAll, setIsSelectedAll] = useState(true);
+  const [isSelected, setIsSelected] = useState({
+    1: false,
+    2: false,
+    3: false,
+    4: false,
+  });
+
+  const handleAllButton = () => {
+    setIsSelectedAll(true);
+    setIsSelected({ 1: false, 2: false, 3: false, 4: false });
+  };
+  const handleButton = i => {
+    setIsSelectedAll(false);
+    setIsSelected({ ...isSelected, [i]: !isSelected[i] });
+  };
+  return (
+    <div className="filter">
+      <p>총 177개의 상품이 있습니다.</p>
+      <div className="buttons">
+        <button
+          onClick={() => {
+            handleAllButton();
+          }}
```

Comment:
이벤트 핸들러로 전달되는 함수의 형태가, handleAllButton 함수를 실행시키는 함수입니다.
이런 경우 handleAllButton 함수 자체를 이벤트 핸들러로 전달하시면 됩니다.
```suggestion
          onClick={handleAllButton}
```